test/core/credentials/transport/tls/spiffe_utils_test.cc

```diff
@@ -255,6 +291,206 @@ TEST(SpiffeId, TripleDotsSuccess) {
+          "certificate: INVALID_ARGUMENT: SPIFFE Bundle key entry has x5c "
+          "field with length != 1. Key entry x5c field MUST have length of "
+          "exactly 1.; field:trust_domains[\"google.com\"].keys[0].x5c "
+          "error:array length must be 1, got 2]"));
+}
+
+TEST(SpiffeBundle, WrongRootFails) {
+  EXPECT_EQ(
+      SpiffeBundleMap::FromFile(
+          "test/core/credentials/transport/tls/test_data/spiffe/test_bundles/"
+          "spiffebundle_wrong_root.json")
+          .status(),
+      absl::InvalidArgumentError("errors validating JSON: [field:trust_domains "
+                                 "error:field not present]"));
+}
+
+TEST(SpiffeBundle, WrongUseFails) {
+  EXPECT_EQ(
+      SpiffeBundleMap::FromFile(
+          "test/core/credentials/transport/tls/test_data/spiffe/test_bundles/"
+          "spiffebundle_wrong_use.json")
+          .status(),
+      absl::InvalidArgumentError(
+          "errors validating JSON: "
+          "[field:trust_domains[\"example.com\"].keys[0].use error:value must "
+          "be \"x509-svid\", got \"NOT-x509-svid\"]"));
+}
+
+TEST(SpiffeBundle, MultipleTrustDomainsSuccess) {
+  constexpr absl::string_view path =
+      "test/core/credentials/transport/tls/test_data/spiffe/test_bundles/"
+      "spiffebundle.json";
+  std::string json_str = testing::GetFileContents(path.data());
+  auto json = JsonParse(json_str);
+  ASSERT_TRUE(json.ok());
+  auto bundle_map = LoadFromJson<SpiffeBundleMap>(*json);
+  ASSERT_TRUE(bundle_map.ok()) << bundle_map.status();
+  ASSERT_EQ(bundle_map->size(), 2);
+  {
+    // check the example.com bundle
+    auto roots = bundle_map->GetRoots("example.com");
+    ASSERT_TRUE(roots.ok());
+    EXPECT_EQ(roots->size(), 1);
+    auto certificate = ReadCertificate((*roots)[0]);
+    ASSERT_TRUE(certificate.ok()) << certificate.status();
+    auto expected_certificate = ReadCertificateFromFile(
+        "test/core/credentials/transport/tls/test_data/spiffe/"
+        "spiffe_cert.pem");
+    ASSERT_TRUE(expected_certificate.ok()) << expected_certificate.status();
+    EXPECT_EQ(X509_cmp(*certificate, *expected_certificate), 0);
+    X509_free(*certificate);
+    X509_free(*expected_certificate);
+  }
+  {
+    // check the test.example.com bundle
+    auto roots = bundle_map->GetRoots("test.example.com");
+    ASSERT_TRUE(roots.ok());
+    EXPECT_EQ(roots->size(), 1);
+    auto certificate = ReadCertificate((*roots)[0]);
+    ASSERT_TRUE(certificate.ok()) << certificate.status();
+    auto expected_certificate = ReadCertificateFromFile(
+        "test/core/credentials/transport/tls/test_data/spiffe/test_bundles/"
+        "server1_spiffe.pem");
+    ASSERT_TRUE(expected_certificate.ok()) << expected_certificate.status();
+    EXPECT_EQ(X509_cmp(*certificate, *expected_certificate), 0);
+    X509_free(*certificate);
+    X509_free(*expected_certificate);
+  }
+}
+
+TEST(SpiffeBundle, MultipleRootsSuccess) {
+  std::string path =
```

Comment:
Similar to above: This should be `constexpr absl::string_view kPath`, and we should use `std::string(kPath)` on the next line.

Reply:
Changed to just code the string in the call itself, no need to hold a variable